File: examples/box.cpp
#include <string>
#include <sstream>
#include <unistd.h>
#include <iostream>

#include "hilma/types/Mesh.h"
#include "hilma/ops/generate.h"
#include "hilma/io/ply.h"

int main(int argc, char **argv) {
    int size = 1024;

    hilma::Mesh mesh = hilma::box(1024, 1024, 1024, 10, 10, 10);
    hilma::savePly("box.ply", mesh, false);

    return 1;
}

File: examples/box.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

from hilma import Mesh, box, savePly
import numpy

size = 1024

mesh = box(size, size, size, 10, 10, 10)
mesh.invertWindingOrder()
savePly("box.ply", mesh, False)